File: leetcode/2000.py
class Solution:

    def reversePrefix(self, word: str, ch: str) -> str:
        pos = None

        for i, c in enumerate(word):
            if c == ch:
                pos = i
                break

        if pos is None:
            return word

        return word[pos::-1] + word[pos + 1:]


if __name__ == "__main__":
    print(Solution().reversePrefix("abcdefd", "d"))

File: leetcode/2000.c
char* reversePrefix(char* word, char ch) {
  int left = 0;
  int right = 0;
  char tmp;

  while (word[right] != '\0' && word[right] != ch) {
    ++right;
  }

  if (word[right] == '\0') {
    return word;
  }

  while (left < right) {
    tmp = word[left];
    word[left] = word[right];
    word[right] = tmp;
    ++left;
    --right;
  }

  return word;
}